### PR TITLE
Normalize Book Shortcode length (3-character)

### PR DESCRIPTION
**Pull request by tbsexton**
Made it possible for all books to be referred to as their common (?) 3-character shortcode. Most are the first 3 letters, though others are non-standard (e.g. Judges->Jdg, Philippians->Php)

- Song of Songs/Solomon is not shortened. What 3-letter shortcode makes sense?
- (1,2,3)John has it's own separated RegEx to unify the shortcodes
-  Regex is order-dependent, so failing to match philippians should make it OK to match `Phi` as Philemon (as I typically see done).

### CODE REVIEW

File: pythonbible/regular_expressions.py [hunk 1]
```diff
@@ -107,24 +108,24 @@ def _add_suffix(regex: str, suffix: str = None) -> str:
...
+    Book.ZECHARIAH: r"Zec\.*(?:h\.*(?:ariah)?)?",
     Book.MALACHI: r"Mal\.*(?:achi)?",
-    Book.MATTHEW: r"Matt\.*(?:hew)?",
-    Book.MARK: "Mark",
-    Book.LUKE: "Luke",
-    Book.JOHN: r"(?<!(?:1|2|3|I)\s)(?<!(?:1|2|3|I))John",
+    Book.MATTHEW: r"Mat\.*(?:t\.*(?:hew)?)?",
+    Book.MARK: r"Mar\.*(?:k)?",
+    Book.LUKE: r"Luk\.*(?:e)?",
+    Book.JOHN: r"(?<!(?:1|2|3|I)\s)(?<!(?:1|2|3|I))"+JOHN_REGULAR_EXPRESSION
```
Comment (avendesora) [inline pythonbible/regular_expressions.py:128]: Looks like this line is missing the comma at the end.

**Review by avendesora — CHANGES_REQUESTED** [review]

File: pythonbible/regular_expressions.py (hunk 1 above)
Comment (tbsexton) [reply pythonbible/regular_expressions.py]: Should be fixed, added `Act\.*(?:s)?` as well.

**Review by avendesora — CHANGES_REQUESTED** [review]
We need to update the Philemon regular expression to pass tests. I think changing line 155 to the following should at least get them passing:

Book.PHILEMON: r"(Philemon|Phile|Phi|Phlm)\.*(?:m\.*(?:on)?)?",

After that, I think there's a little bit of tweaking that'll need to be done to preserve 100% test coverage.

File: pythonbible/regular_expressions.py [hunk 2]
```diff
@@ -152,7 +152,7 @@ def _add_suffix(regex: str, suffix: str = None) -> str:
     ),
     Book.TITUS: r"Tit\.*(?:us)?",
     # assume 'Phi' is Philemon if Philippians failed?
-    Book.PHILEMON: r"(Phi|Phlm|Phile)\.*(?:m\.*(?:on)?)?", 
+    Book.PHILEMON: r"(Philemon|Phile|Phi|Phlm).(?:m.(?:on)?)?",
```
Comment (avendesora) [inline pythonbible/regular_expressions.py:155]: Maybe a better, or at least more readable, regex for Philemon would be:

Book.PHILEMON: r"(Philemon|Philem\.*|Phile\.*|Phlm\.*|Phi\.*)",

We've got one test currently failing because of the "."

**Review by avendesora — CHANGES_REQUESTED** [review]

File: pythonbible/regular_expressions.py (hunk 2 above)
Comment (avendesora) [reply pythonbible/regular_expressions.py:155]: I think the formatting of my previous review messed up the regular expression that I was recommending. Sorry about that.

Comment (tbsexton) [reply pythonbible/regular_expressions.py]: NP, I went ahead and committed your comment, awaiting the test suite. Seems nice to be more explicit for this edge-case, anyway.

**Review by avendesora — APPROVED** [review]

**Comment by avendesora** [comment]
@tbsexton Thanks for the contribution!

**Comment by tbsexton** [comment]
> The "Test" GitHub action failed due to it failing to upload the code coverage report to codacy. The actual tests all passed with 100% coverage.

Cool, yeah I had replicated a dev environment locally to verify as well (to rule out tests). Thanks!
